src/recipes/engine-singleton.md

```diff
@@ -81,16 +82,37 @@ unsafe impl ExtensionLibrary for MyExtension {
 
     fn on_level_deinit(level: InitLevel) {
         if level == InitLevel::Scene {
-            // Unregistering is needed to avoid memory leaks and 
-            // warnings, especially for hot reloading.
-            Engine::singleton().unregister_singleton(
-                StringName::from("MyEditorSingleton")
-            );
+            // Get the `Engine` instance and `StringName` for your singleton.
+            let mut engine = Engine::singleton();
+            let singleton_name = StringName::from("MyEditorSingleton");
+
+            // We need to retrieve the pointer to the singleton object,
+            // as it has to be freed manually - unregistering singleton 
+            // doesn't do it automatically.
+            let singleton = engine
+                .get_singleton(singleton_name.clone())
+                .expect("cannot retrieve the singleton");
+
+            // Unregistering singleton and freeing the object itself is needed 
+            // to avoid memory leaks and warnings, especially for hot reloading.
+            engine.unregister_singleton(singleton_name);
+            singleton.free();
         }
     }
 }
 ```
 
+```admonish warning title="Singletons inheriting from *RefCounted*"
+It is recommended to use manually-managed class as a base (often `Object` will be enough) for custom singletons. Godot engine isn't
+managing the instances registered as singletons on its own, so it's very possible that `RefCounted`-inheriting one will be dropped
+prematurely, leading to errors on its retrieval.
+
+If for any reason you need to have instance of reference-counted object registered as singleton, this [issue thread][refcounted-singleton-issue]
+presents some possible workarounds.
+```
```

Comment:
This is not generally true; you can register singletons in GDScript just fine. It's likely a bug on our end or in the GDExtension API.

Please consider my previous comment:

> I would link directly to https://github.com/godot-rust/gdext/issues/522 and straight discourage `RefCounted` objects from being used as singletons for the moment.

Also, two general style issues:
- Your sentences miss a lot of articles ("a", "an", "the").
- Let's not speculate -- it's not "very possible" that `RefCounted` ones are dropped prematurely, this currently is _known_ to happen (due to a bug).

So, to keep this warning short, I'd simply higlight the facts:
- Only manually managed objects are currently supported, no classes inheriting from `RefCounted`.
- See issue #522 for more information (including workarounds).

Everything else is explained in the issue. If the information updates, there's one less place to keep in sync.

---

Sorry if this all sounds nitpicky, but it's important that we are a) concise and b) depict an accurate view of the status quo, including limitations.